src/Panel.tsx

```diff
@@ -1,164 +1,129 @@
-        >
-          {iconNode}
-        </div>
-      )
-    );
   };
 
-  renderTitle = () => {
-    const { header, prefixCls, collapsible } = this.props;
-
-    return (
-      <span
-        className={`${prefixCls}-header-text`}
-        onClick={collapsible === 'header' ? this.onItemClick : null}
+  // ======================== Icon ========================
+  let iconNode = typeof expandIcon === 'function' ? expandIcon(props) : <i className="arrow" />;
+  if (iconNode) {
+    iconNode = (
+      <div
+        className={`${prefixCls}-expand-icon`}
+        onClick={['header', 'icon'].includes(collapsible as string) ? handleItemClick : undefined}
       >
-        {header}
-      </span>
-    );
-  };
-
-  render() {
-    const {
-      className,
-      id,
-      style,
-      prefixCls,
-      headerClass,
-      children,
-      isActive,
-      destroyInactivePanel,
-      accordion,
-      forceRender,
-      openMotion,
-      extra,
-      collapsible,
-      ...rest
-    } = this.props;
-
-    const disabled = collapsible === 'disabled';
-    const collapsibleHeader = collapsible === 'header';
-    const collapsibleIcon = collapsible === 'icon';
-
-    const itemCls = classNames(
-      {
-        [`${prefixCls}-item`]: true,
-        [`${prefixCls}-item-active`]: isActive,
-        [`${prefixCls}-item-disabled`]: disabled,
-      },
-      className,
+        {iconNode}
+      </div>
     );
+  }
 
-    const headerCls = classNames(`${prefixCls}-header`, {
-      [headerClass]: headerClass,
-      [`${prefixCls}-header-collapsible-only`]: collapsibleHeader,
-      [`${prefixCls}-icon-collapsible-only`]: collapsibleIcon,
-    });
-
-    /** header 节点属性 */
-    const headerProps: React.HTMLAttributes<HTMLDivElement> = {
-      className: headerCls,
-      'aria-expanded': isActive,
-      'aria-disabled': disabled,
-      onKeyPress: this.handleKeyPress,
-    };
-
-    if (!collapsibleHeader && !collapsibleIcon) {
-      headerProps.onClick = this.onItemClick;
-      headerProps.role = accordion ? 'tab' : 'button';
-      headerProps.tabIndex = disabled ? -1 : 0;
-    }
-
-    const ifExtraExist = extra !== null && extra !== undefined && typeof extra !== 'boolean';
+  const collapsePanelClassNames = classNames(
+    {
+      [`${prefixCls}-item`]: true,
+      [`${prefixCls}-item-active`]: isActive,
+      [`${prefixCls}-item-disabled`]: disabled,
+    },
+    className,
+  );
+
+  const headerClassName = classNames({
+    [`${prefixCls}-header`]: true,
+    headerClass,
+    [`${prefixCls}-header-collapsible-only`]: collapsibleHeader,
+    [`${prefixCls}-icon-collapsible-only`]: collapsibleIcon,
+  });
+
+  // ======================== HeaderProps ========================
+  const headerProps: React.HTMLAttributes<HTMLDivElement> = {
+    className: headerClassName,
+    'aria-expanded': isActive,
+    'aria-disabled': disabled,
+    onKeyPress: handleKeyPress,
+  };
```

Comment:
`onKeyPress` is deprecated, do I need to make adjustments? [Element: keypress event](https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event)

Reply:
we can change to `onKeydown`.

Reply:
Okay, I'll try it.